File: 0050-powx-n/0050-powx-n.cpp
class Solution {
public:
    
    double helper(double x,int n)
    {        
        if(n==0)
            return 1;

        double val=helper(x,n/2);

        if(n&1)
            return x*val*val;
        
        return val*val;
    }
    
    double myPow(double x, int n) {
        if(n<0)
            return helper(1/x,abs(n));
        return helper(x,n);
    }
};

File: 0050-powx-n/0050-powx-n.java
class Solution {
    private double pow(double x,int n)
    {
        if(n==0)
            return 1;
        if(n==1)
            return x;
        double val = pow(x,n/2);
        return (n%2)==0?val*val:val*val*x;
    }
    public double myPow(double x, int n) {
        if(n<0)
            return 1/pow(x,-n);
        return pow(x,n);
    }
}